clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -222,17 +228,24 @@ public RecordAppendResult append(TopicPartition tp,
         }
     }
 
+    private MemoryRecordsBuilder getRecordsBuilder(ByteBuffer buffer, byte maxUsableMagic) {
```

Comment:
Nit: this could just be `recordsBuilder`. We generally avoid the `get` prefix in Kafka.